app/actions/battles.py

```diff
@@ -0,0 +1,5 @@
+def duel(knight1: dict, knight2: dict) -> dict:
+    knight1["hp"] += knight1["add_some_abilities"]
+    knight2["hp"] += knight2["add_some_abilities"]
+    knight1["hp"] -= knight2["power"]
+    knight2["hp"] -= knight1["power"]

```

Comment:
Consider checking if 'power' exists in the knight dictionaries before using it to avoid potential KeyError.